src/quickunlock/Polkit.cpp

```diff
@@ -82,18 +83,37 @@ bool Polkit::getKey(const QString& dbPath, QByteArray& key)
     if (!hasKey(dbPath))
         return false;
 
+    pid_t app_pid = QCoreApplication::applicationPid();
+
+    QString processStatPath = QString("/proc/%1/stat").arg(app_pid);
+    QFile processStatFile(processStatPath);
+
+    if(!processStatFile.open(QIODevice::ReadOnly | QIODevice::Text)){
+        qDebug() << "polkit: failed to open " << processStatPath << " for reading";
+        return false;
+    }
+
+    QTextStream processStatStream(&processStatFile);
+    QString processStatInfo = processStatStream.readLine();
+    processStatFile.close();
+
+    QString processStartTime = processStatInfo.split(" ")[21];
```

Comment:
May want to move this to a helper function in [`gui/osutils/nixutils`](https://github.com/keepassxreboot/keepassxc/tree/develop/src/gui/osutils/nixutils).